keanu-python/keanu/vertex/base.py

```diff
@@ -182,22 +191,29 @@ def __parse_arg(arg: Union[vertex_constructor_param_types, shape_types]) -> Java
             return arg.unwrap()
         elif isinstance(arg, collections.Collection) and all(isinstance(x, runtime_primitive_types) for x in arg):
             return k.to_java_long_array(arg)
-        elif isinstance(arg, runtime_primitive_types) or isinstance(arg, JavaObject):
+        elif isinstance(arg, (runtime_primitive_types, JavaObject, runtime_str_types)):
             return arg
         else:
             raise ValueError("Can't parse generic argument. Was given {}".format(type(arg)))
 
+    @staticmethod
+    def _from_java_vertex(java_vertex: JavaObject) -> 'Vertex':
+        return Vertex(java_vertex, None)
+
     @staticmethod
     def _to_generator(java_vertices: Union[JavaList, JavaArray]) -> Iterator['Vertex']:
-        return (Vertex(java_vertex) for java_vertex in java_vertices)
+        return (Vertex._from_java_vertex(java_vertex) for java_vertex in java_vertices)
 
     @staticmethod
     def _get_python_id(java_vertex: JavaObject) -> Tuple[JavaObject, ...]:
         return tuple(java_vertex.getId().getValue())
 
     @staticmethod
-    def _get_python_label(java_vertex: JavaObject) -> str:
-        return java_vertex.getLabel().getQualifiedName()
+    def _get_python_label(java_vertex: JavaObject) -> Optional[str]:
+        try:
+            return java_vertex.getLabel().getQualifiedName()
+        except (AttributeError, Py4JError):
```

Comment:
Looking at the Java code it's not clear to me that this would throw an exception? Unless a label was created with a null name, but would/should that be the case?

Reply:
label is null by default

Reply:
removed it!